verticapy/performance/vertica/collection/collection_tables.py

```diff
@@ -0,0 +1,584 @@
+
+    def get_import_name(self) -> str:
+        return self._get_import_name(fully_qualified=False)
+
+    def get_super_proj_name_fq(self) -> str:
+        return f"{self._get_import_name(fully_qualified=True)}_super"
+
+    def _get_import_name(self, fully_qualified) -> str:
+        return (
+            f"{self.schema}.{self.import_prefix}{self.name}{self.import_suffix}"
+            if fully_qualified
+            else f"{self.import_prefix}{self.name}{self.import_suffix}"
+        )
+
+    def get_staging_name(self) -> str:
+        return f"{self.schema}.{self.import_prefix}{self.name}{self.staging_suffix}{self.import_suffix}"
+
+    def get_drop_table_sql(self) -> str:
+        # Drop table can be in the base class because it is
+        # DROP TABLE IF EXISTS {self.get_import_name_fq}
+        raise NotImplementedError(f"Drop table not implemented yet")
+
+    # Recall: abstract methods won't raise by default
+    @abstractmethod
+    def get_create_table_sql(self) -> str:
+        raise NotImplementedError(
+            f"get_create_table_sql is not implemented in the base class CollectionTable."
+            f" Current table name = {self.name} schema {self.schema}"
+        )
+
+    @abstractmethod
+    def get_create_projection_sql(self) -> str:
+        raise NotImplementedError(
+            f"get_create_projection_sql is not implemented in the base class CollectionTable"
+            f" Current table name = {self.name} schema {self.schema}"
+        )
+
+    @abstractmethod
+    def has_copy_staging(self) -> str:
+        raise NotImplementedError(
+            f"get_create_projection_sql is not implemented in the base class CollectionTable"
+            f" Current table name = {self.name} schema {self.schema}"
+        )
+
+    @abstractmethod
+    def copy_from_local_file(self, filename: str) -> str:
+        # This method will generate and run the copy statement
+        # It will copy to a staging table when necessary
+        raise NotImplementedError(f"")
+
+
+class AllTableNames(Enum):
+    COLLECTION_EVENTS = "collection_events"
+    COLLECTION_INFO = "collection_info"
+    DC_EXPLAIN_PLANS = "dc_explain_plans"
+    DC_QUERY_EXECUTIONS = "dc_query_executions"
+    DC_REQUESTS_ISSUED = "dc_requests_issued"
+    EXECUTION_ENGINE_PROFILES = "execution_engine_profiles"
+    EXPORT_EVENTS = "export_events"
+    HOST_RESOURCES = "host_resources"
+    QUERY_CONSUMPTION = "query_consumption"
+    QUERY_PLAN_PROFILES = "query_plan_profiles"
+    QUERY_PROFILES = "query_profiles"
+    RESOURCE_POOL_STATUS = "resource_pool_status"
+
+
+def getAllCollectionTables(
+    target_schema: str, key: str
+) -> Mapping[str, CollectionTable]:
+    result = {}
+    for name in AllTableNames:
+        name_str = name.value
+        c = collectionTableFactory(name_str, target_schema, key)
+        result[name.name] = c
+
+    return result
+
+
+def collectionTableFactory(
+    table_name: str, target_schema: str, key: str
```

Comment:
You can specify the type of table_name to be  AllTableNames if this function is not supposed to work with any random table name.
If you do that you'll need `.value` neither in the getAllCollectionTables function nor in the following if statements.

Reply:
Good point. 

Based on this feedback, I decided to update the usage of the enum. We really do want to compare to the enum. I added a field to the base class CollectionTable named table_type. The table_type stores the enum. Then we can avoid using enum.value in sprinkled around in many places.